ndonnx/additional/_additional.py

```diff
@@ -147,6 +148,15 @@ def make_nullable(x: Array, null: Array) -> Array:
     return out
 
 
+def _getitem(x: Array, index: IndexType) -> ndx.Array:
```

Comment:
I'm considering making `_getitem` and `_static_shape` public free functions in the future and wanted to leave it as a private function in the interim.

Reply:
Under what circumstance would I prefer to call `_getitem(array, index)` over `array[index]`?

Reply:
I agree with @adityagoel4512 on this choice - it makes sense to strip down as much as possible from the `Array` implementation and put it into free functions even if it adds a layer of indirection.

Reply:
Fair enough. Let's leave these functions as they are right now, since that leaves the implementations in one place. I am not quite convinced that it is helpful to make them public eventually, though.